perfact/zodbsync/extedit.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import json
+
+from . import helpers
+from .zodbsync import mod_read, mod_write
+
+
+def launch(context, script, path, source=None, orig_source=None):
+    '''
+    Launcher for external edit.
+
+    If called without a source, it is used to create a control file that
+    contains the authentication header found in the current request and the
+    content of the file that can be found under path.
+
+    If a source is provided, updates the object given by path, but only if the
+    current source matches that given in orig_source.
+
+    A wrapper script should be placed in the top-level of the ZODB that is only
+    accessible for Manager and delegates to this.
+    '''
+
+    resp = context.REQUEST.RESPONSE
+
+    if source is None:
+        content_type = 'application/x-perfact-zopeedit'
+        result = controlfile(
+            context=context,
+            path=path,
+            url=script.absolute_url(),
+        )
+    else:
+        content_type = 'application/json'
+        result = update(
+            context=context,
+            path=path,
+            source=source,
+            orig_source=orig_source,
+        )
+        result = json.dumps(result)
+
+    resp.setHeader('Content-Type', content_type)
+    return result
+
+
+def find_obj(context, path):
+    '''
+    Locate object at given path
+    '''
```

Comment:
Perhaps for a future version: Show an error essage for a missing object (in case that someone deleted or renamed it).